src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -2205,8 +2205,9 @@ class UpdateStatementSegment(BaseSegment):
     match_grammar = StartsWith("UPDATE")
     parse_grammar = Sequence(
         "UPDATE",
-        Ref("TableReferenceSegment"),
+        Ref("AliasedTableReferenceSegment"),
```

Comment:
What's the outcome of changing this line to:
```suggestion
        OneOf(Ref("TableReferenceSegment"), Ref("AliasedTableReferenceSegment"))
```

I think in its current state even un-aliased table reference will be parsed as one. This should solve my comment here https://github.com/sqlfluff/sqlfluff/pull/865/files#r597557373

Reply:
I'll make that change. I think I was so focused on getting a successful parse, I didn't think about cleanliness. I wonder if we should consider (later, not this PR) changing `AliasedTableReferenceSegment` so the alias is required. The naming is a bit odd, calling it "aliased" when the alias is optional. 🤔

Reply:
Ah yes I definitely think so. Feel free to do it here or we can tackle separately

Reply:
I hadn't realised the alias was optional in that segment